### PR TITLE
Add check RHEL kernel

### PR DESCRIPTION
**Pull request by fellipeh**
If there is no RHEL kernel, the convert process will halt with CRITICAL message.

Signed-off-by: Fellipe Henrique <fpedrosa@redhat.com>

### CODE REVIEW

File: convert2rhel/pkghandler.py [hunk 1]
```diff
@@ -356,6 +357,11 @@ def replace_non_red_hat_packages():
 def preserve_only_rhel_kernel():
     loggerinst = logging.getLogger(__name__)
```
Comment (ZhukovGreen) [inline convert2rhel/pkghandler.py:358]: Please instantiate the logger outside of the function scope, you probably don't want it re-instantiate each time you call this function. Pls see the following log example, when you have logger outside
```
----------------------------- live log collection ------------------------------
| 10:43:49 | my_pkg | INFO | __init__.py | Logger configured
| 10:43:49 | my_pkg.log | INFO | log.py | Logger obtained
collected 1 item

test_logger.py::test_foo 
-------------------------------- live log call ---------------------------------
| 10:43:49 | my_pkg.log | INFO | log.py | Foo called
| 10:43:49 | my_pkg.log | INFO | log.py | Foo called
PASSED
```

Comment (ZhukovGreen) [reply convert2rhel/pkghandler.py:358]: /cc @SpyTec

Comment (SpyTec) [reply convert2rhel/pkghandler.py:358]: Outside the scope for this and #60, should be fixed in a separate PR.

As you [linked in #60](https://github.com/oamg/convert2rhel/pull/60#discussion_r432329984), Logging Cookbook mentions 

> Multiple calls to logging.getLogger('someLogger') return a reference to the same logger object.

If the method call returns a singleton I'm not sure I understand the reason for instantiating it outside method scope

Comment (ZhukovGreen) [reply convert2rhel/pkghandler.py:358]: @SpyTec because it is an additional overhead which still returns the same object

Comment (bocekm) [reply convert2rhel/pkghandler.py:358]: See https://github.com/oamg/convert2rhel/pull/60#discussion_r432385516. I agree with changing the way we work with the logger now. But let's do that under a separate PR for all the code.

**Comment by fellipeh** [comment]
Fixed conflict and rebased

File: run_unit_tests.sh [hunk 2]
```diff
@@ -27,7 +27,7 @@
 
 # -- TEST OUTPUT --
 # To let the tests print to stdout, uncomment the following line:
-#TEST_OUTPUT="--nocapture"
+TEST_OUTPUT="--nocapture"
```
Comment (SpyTec) [inline run_unit_tests.sh:30]: Committed by mistake? Otherwise seems out of scope

File: convert2rhel/unit_tests/pkghandler_test.py [hunk 3]
```diff
@@ -28,6 +28,12 @@
 
 
 class TestPkgHandler(unit_tests.ExtendedTestCase):
+    class GetInstalledPkgsWithFingerprintMocked(unit_tests.MockFunction):
+        def __init__(self, data=None):
+            self.data = data
+
+        def __call__(self, *args, **kwargs):
+            return self.data
```
Comment (SpyTec) [inline convert2rhel/unit_tests/pkghandler_test.py:36]:
```suggestion
    class GetInstalledPkgsWithFingerprintMocked(unit_tests.MockFunction):
        def __init__(self, data=None):
            self.data = data
            self.called = 0

        def __call__(self, *args, **kwargs):
            self.called += 1
            return self.data
```

File: convert2rhel/unit_tests/pkghandler_test.py [hunk 4]
```diff
@@ -380,12 +388,13 @@ def __call__(self):
     @unit_tests.mock(pkghandler, "install_gpg_keys",
                      DumbCallableObject())
     @unit_tests.mock(utils, "run_subprocess", RunSubprocessMocked())
+    @unit_tests.mock(pkghandler, "get_installed_pkgs_by_fingerprint",
+                     GetInstalledPkgsWithFingerprintMocked(data=['kernel']))
     def test_preserve_only_rhel_kernel(self):
         pkghandler.preserve_only_rhel_kernel()
 
         self.assertEqual(utils.run_subprocess.cmd, "yum update -y kernel")
```
Comment (SpyTec) [inline convert2rhel/unit_tests/pkghandler_test.py:396]:
```suggestion
    def test_preserve_only_rhel_kernel(self):
        pkghandler.preserve_only_rhel_kernel()

        self.assertEqual(utils.run_subprocess.cmd, "yum update -y kernel")
        self.assertEqual(pkghandler.get_installed_pkgs_by_fingerprint.called, 1)
```

Comment (SpyTec) [inline convert2rhel/unit_tests/pkghandler_test.py:396]: Create another test for `preserve_only_rhel_kernel()` to verify that branch for `loggerinst.critical` works as well

File: convert2rhel/unit_tests/pkghandler_test.py [hunk 5]
```diff
@@ -577,6 +586,16 @@ def test_install_additional_rhel_kernel_pkgs(self):
         pkghandler.install_additional_rhel_kernel_pkgs(removed_pkgs)
         self.assertEqual(pkghandler.call_yum_cmd.called, 2)
 
+    @unit_tests.mock(pkghandler, "get_installed_pkgs_by_fingerprint",
+                     GetInstalledPkgsWithFingerprintMocked(data=['kernel']))
+    def test_check_exist_rhel_kernel_by_fingerprint(self):
```
Comment (SpyTec) [inline convert2rhel/unit_tests/pkghandler_test.py:591]:
```suggestion
    def test_check_installed_rhel_kernel_returns_true(self):
```

File: convert2rhel/unit_tests/pkghandler_test.py [hunk 6]
```diff
@@ -577,6 +586,16 @@ def test_install_additional_rhel_kernel_pkgs(self):
...
+        self.assertEqual(pkghandler.is_rhel_kernel_installed(), True)
+
+    @unit_tests.mock(pkghandler, "get_installed_pkgs_by_fingerprint",
+                     GetInstalledPkgsWithFingerprintMocked(data=[]))
+    def test_check_exist_rhel_kernel_by_fingerprint(self):
```
Comment (SpyTec) [inline convert2rhel/unit_tests/pkghandler_test.py:596]:
```suggestion
    def test_check_installed_rhel_kernel_returns_false(self):
```

File: convert2rhel/unit_tests/pkghandler_test.py (hunk 4 above)
Comment (SpyTec) [reply convert2rhel/unit_tests/pkghandler_test.py:398]: Could probably rewrite that to be more clear. There isn't any tests right now covering what happens inside `preserve_only_rhel_kernel()` if `is_rhel_kernel_installed()` returns False

Can be checked with a new test mocking out `is_rhel_kernel_installed` to return False and assert `self.assertRaises(SystemExit, preserve_only_rhel_kernel)`

**Comment by fellipeh** [comment]
Code rebased and squashed
